shared/src/vyxal/ListHelpers.scala

```diff
@@ -854,6 +854,21 @@ object ListHelpers:
     current
   end reduce
 
+  def reshape(iterable: VList, shape: Seq[VNum]): VAny =
+    var index = -1
```

Comment:
Why not use a `iterable.iterator` here? If the backing list is a linked list, indexing into it over and over again could be inefficient. I'm assuming an iterator would be more performant. When you hit the end of the iterator, you could make a new iterator since you have the original `VList` available